### PR TITLE
Expose ENet statistics and throttle configuration

### PR DESCRIPTION
**Pull request by AndreaCatania**
- Exposed ENet stats
- Exposed ENet throttle config

Useful to obtain stats like:
```
- Loss: 0.112915%  
- Loss variance: 0.192261%  
- Ping: 15ms  
- Ping variance: 15ms  
- Throttle: 32
- Total sent data: 682202bytes 
- Total received data: 682202bytes
```

This work has been kindly sponsored by IMVU.

### CODE REVIEW

**Comment by akien-mga** [comment]
That's a lot of methods, it will make the API much harder to read IMO.

I'd suggest grouping part or all of this stuff together in one or more functions that return a Dictionary (like `Engine.get_version_info()`) instead of one method for each metric.

But @Faless may see it differently, so let's wait for his feedback.

Once agreed upon, the docs should also be written for the new methods.

**Comment by AndreaCatania** [comment]
Not all stats are in all circumstances needed, and group them mean just add overhead:

A system may just want to read the packet loss, another one may just need to read the RTT and so on. I would like to not fetch always all data to just read 1 element.

On the other hand, I could read all in 1 place and then spit using GDScript, but this would just make the GDScript code bloated and hard to maintain.

**Comment by aaronfranke** [comment]
@AndreaCatania Is this still desired? If so, it needs to be [rebased](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) and tested on the latest master branch. While there are no conflicts, rebasing is important so that reviewers can easily test the PR.

**Comment by Faless** [comment]
I don't like this PR as is, as @akien-mga said this is way too complex.
If we want to provide more low level access to ENet we should have a dedicated classes for it (e.g. `ENetServer` [better name needed] and `ENetPacaketPeer`), and probably let the High level multiplayer implementation use that instead.
Or, group them together. And why do we need the `pop` functions at all?

**Comment by reduz** [comment]
I think this is a very good idea, but instead of having all individual methods, I suggest using an enum, so you can do like:

```
float packet_loss = enet.get_statistic(Enet.STATISTIC_PACKET_LOSS)
```

naming could probably be better but I think this is most likely the best way to implement.

**Comment by AndreaCatania** [comment]
Yes, it works too.

**Comment by mhilbrunner** [comment]
Networking meeting: merge this if reduz's suggestion is implemented. Looks good otherwise!
